### PR TITLE
Setup tests with synthetic data

### PR DESCRIPTION
**Pull request by akkuldn**
Have added tests to validate data from /survey/admin/QElo_filter_response endpoint using synthetic data

### CODE REVIEW

File: tests/test_responses_between_dates.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    - Date range 1 year prior to the date where test data is stored
+    - Current date
+    - Date range covers first week of test data
+    - Date range covers first two weeks of test data
+    - Date range from a week prior to the dates where test data is present ,to the week succeeding the last date which has the test date
+"""
+import json
+import os
+import pytest
+from unittest.mock import patch
```
Comment (qxf2) [inline tests/test_responses_between_dates.py:14]: Import order is not right. Look up pep-8 or put as a linter.
Remove modules that are not needed.

File: tests/test_responses_between_dates.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
+"""
+Tests for survey API endpoint that fetches all the user responses between two given dates 
+Test API response for different date ranges
```
Comment (qxf2) [inline tests/test_responses_between_dates.py:3]: Add the API endpoint here.

File: tests/test_responses_between_dates.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    - Date range whithin which all the test data is present
```
Comment (qxf2) [inline tests/test_responses_between_dates.py:4]: *Whithin has a typo

File: tests/test_responses_between_dates.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+from decouple import config
+from urllib.parse import urljoin
+from datetime import datetime , timedelta, date
+from dateutil.relativedelta import relativedelta, FR
+
+URL = config("URL")
+API_KEY = config("API_KEY")
+
+#Get the url for the API endpoint that fetches user responses b/w two given dates
+responses_url = urljoin(URL, "survey/admin/QElo_filter_response")
```
Comment (qxf2) [inline tests/test_responses_between_dates.py:25]: Global variables are usually in CAPS.

File: tests/test_responses_between_dates.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+            #Date range covers first two weeks of test data
+            ({"start_date": "1969-12-29","end_date": "1970-01-09"},[
+            {"respondent_id": 1, "date": "1970-01-02", "question_no": 2, "answer": "Stingy Taker"},
+            {"respondent_id": 2, "date": "1970-01-02", "question_no": 1, "answer": "Generous Giver"},
+            {"respondent_id": 2, "date": "1970-01-02", "question_no": 2, "answer": "Generous Giver"},
+            {"respondent_id": 1, "date": "1970-01-09", "question_no": 2, "answer": "Stingy Taker"},
+            {"respondent_id": 2, "date": "1970-01-09", "question_no": 1, "answer": "Generous Giver"}
+            ]),
+
+            #Date range from a week prior to the dates where test data is present ,to the week succeeding the last date which has the test date
```
Comment (qxf2) [inline tests/test_responses_between_dates.py:68]: Is this line too long? Ask your IDE.

File: tests/test_responses_between_dates.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+
+@pytest.mark.parametrize("date_range,expected_response", TEST_DATA)
+def test_data_between_dateranges(date_range,expected_response):
+    "Test the response data recieved for different date ranges"
+    
+    #Get the response from API
+    data_date_range = date_range
+    json_data = json.dumps(data_date_range)
+    response = requests.post(responses_url, data = json_data, headers = {'User': API_KEY})
```
Comment (qxf2) [inline tests/test_responses_between_dates.py:89]: Do we need lines 87, 88? Can't we just `data = json.dumps(date_range)` over here.

File: tests/test_responses_between_dates.py [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    response_data = response.json()
+    #Compare the two response recieved with the actual response
+    diff = [i for i in response_data + expected_response if i not in response_data or i not in expected_response]
+    result = len(diff) == 0
+
+    assert response.status_code == 200
+    assert len(response_data) == len(expected_response)
+    assert result == True,f'There are {len(diff)} differences:\n{diff[:5]}'
```
Comment (qxf2) [inline tests/test_responses_between_dates.py:97]: Why is there a slice of diff here? The difference should just be {diff} and not {diff[:5]}

**Review by qxf2 — CHANGES_REQUESTED** [review]
Code review done, please feel free to fix and merge.

File: tests/test_responses_between_dates.py (hunk 1 above)
Comment (akkuldn) [reply tests/test_responses_between_dates.py]: Removed the unwanted imports and rearranged the order

File: tests/test_responses_between_dates.py (hunk 2 above)
Comment (akkuldn) [reply tests/test_responses_between_dates.py:3]: Added

File: tests/test_responses_between_dates.py (hunk 3 above)
Comment (akkuldn) [reply tests/test_responses_between_dates.py]: Corrected the typo

File: tests/test_responses_between_dates.py (hunk 4 above)
Comment (akkuldn) [reply tests/test_responses_between_dates.py]: changed the case

File: tests/test_responses_between_dates.py (hunk 5 above)
Comment (akkuldn) [reply tests/test_responses_between_dates.py]: put it into two lines

File: tests/test_responses_between_dates.py (hunk 6 above)
Comment (akkuldn) [reply tests/test_responses_between_dates.py]: fixed

File: tests/test_responses_between_dates.py (hunk 7 above)
Comment (akkuldn) [reply tests/test_responses_between_dates.py]: Yes you are right, it only prints the first 5 differences, have changed it

**Pull request merged**